### PR TITLE
NFC: tweak lua53 byte extraction

### PR DESCRIPTION
**Pull request by nwf**
This is a first, very small, very tentative step in divorcing our Lua runtime from the ESP8266 / Xtensa architecture.  In particular, rather than using hard-coded offsets when defining byte extractor functions, we rely on the C compiler to compute offsets for us.  I've tried to make this something that might plausibly go upstream, too (and have begun work on the other direction of "meet in the middle", making a patch set against an upstream Lua 5.3, with the goal of reaching parity with `app/lua53`).

This PR should be completely cosmetic, generating identical assembler (but different DWARF, naturally).

- [x] This PR is for the `dev` branch rather than for the `release` branch.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [x] I have thoroughly tested my contribution.
- [x] The code changes are reflected in the documentation at `docs/*`.

### CODE REVIEW

File: app/lua53/luaconf.h [hunk 1]
```diff
@@ -204,6 +204,41 @@
...
+         , "i"((offsetof(type, field)/4)*4) \
+         , "i"((offsetof(type, field)%4)*8) \
+       : );\
+  return res; }
+#endif
+
+#if !defined(LUA_LOAD_BYTE_FN)
+#define LUA_LOAD_BYTE_FN(fn, type, field) \
+  static inline lu_byte fn(const type *o) { \
+    return ((const type *)o)->field; \
```
Comment (HHHartmann) [inline app/lua53/luaconf.h:238]: The cast is not needed here anymore

**Review by HHHartmann — COMMENTED** [review]
Overall a win for code readability. 
More typing, fewer magic numbers, robustness against changed structures and better naming. 
Can't say anything about the asm part. 
Haven't actually tested it. 

Would be nice to push the many casts to `struct GCObject` in the surrounding code but that's another story. 

Looks good to me

**Comment by nwf** [comment]
This seems to survive local testing!

**Comment by HHHartmann** [comment]
Is this ready to go,

**Comment by nwf** [comment]
It's good by me.  Want to hit Rebase and merge?

**Pull request merged**